src/webhook/cached-verifier.ts

```diff
@@ -0,0 +1,62 @@
+import nacl from "tweetnacl";
+import { logger } from "../logger.js";
+
+export class CachedVerifier {
+  private expirationTimes: { [message: string]: number } = {};
+
+  constructor(public knownPublicKeys: string[]) {}
+
+  public verify(signature: string): boolean {
+    if (!this.expirationTimes[signature]) {
```

Comment:
These checks should be moved their own methods

`isSignatureExpired(signature: string)`

Then after verify message

Could return different error messages based on the error

400 BadRequest vs. 401 Unauthorized